modules/hostap/src/supp_api.c

```diff
@@ -906,10 +955,151 @@ int supplicant_channel(const struct device *dev, struct wifi_channel_info *chann
 }
 
 #ifdef CONFIG_AP
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
+int hapd_state(const struct device *dev, int *state)
+{
+	struct hostapd_iface *iface;
+	int ret = 0;
+
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+
+	iface = get_hostapd_handle(dev);
+	if (!iface) {
+		wpa_printf(MSG_ERROR, "Device %s not found", dev->name);
+		ret = -1;
+		goto out;
+	}
+
+	*state = iface->state;
+
+out:
+	k_mutex_unlock(&wpa_supplicant_mutex);
+	return ret;
+}
+
+int hapd_config_network(struct hostapd_iface *iface,
+			struct wifi_connect_req_params *params)
+{
+	int ret = 0;
+
+	if (!hostapd_cli_cmd_v("set ssid %s", params->ssid)) {
+		goto out;
+	}
```

Comment:
you can use a macro do the NULL check + goto out, like we did for WPA supplicant.